### PR TITLE
Sling 3270

### PR DESCRIPTION
**Pull request by JEBailey**
Adding a new non-validating parser to Support HTML5 as well as some additional classes to utilize this parser in other classes other then via the HTMLParser interface

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -37,25 +37,36 @@
...
-      <tag>HEAD</tag>
-  </scm>
+        <tag>HEAD</tag>
+    </scm>
 
     <build>
         <plugins>
             <plugin>
-                <groupId>biz.aQute.bnd</groupId>
-                <artifactId>bnd-maven-plugin</artifactId>
```
Comment (rombert) [inline pom.xml]: Why did you change from the bnd-maven-plugin to the maven-bundle-plugin?

File: src/main/java/org/apache/sling/commons/html/impl/parser/ParseException.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
+/* Generated By:JavaCC: Do not edit this line. ParseException.java Version 5.0 */
```
Comment (rombert) [inline src/main/java/org/apache/sling/commons/html/impl/parser/ParseException.java]: Do we need to commit the generated code to source control? I think there are options to generate it at build time using Maven and then included in the packaged jar, without adding it to source control.

File: pom.xml (hunk 1 above)
Comment (JEBailey) [reply pom.xml]: The baselining seems to be working differently between the two plugins. I'm still gathering information between other tasks, but it looks like the bnd version isn't taking the baseline from the bundle name rather than the package-info.

File: src/main/java/org/apache/sling/commons/html/impl/parser/ParseException.java (hunk 2 above)
Comment (JEBailey) [reply src/main/java/org/apache/sling/commons/html/impl/parser/ParseException.java]: Two things here, first code generation should be part of the build process. When I moved the code into this bundle I missed that plugin and I have since added it.

Yes. If you don't commit your generated code to source control, then there's no guarantee that you will be able to regenerate that exact code. Although maven is awesome compared to what it was before, I have seen jars updated on the backend without changing the version, I've all seen jars disappear. Although I trust maven central a little bit more then an independent repository. I believe in erring on the side of caution.

I also find it easier to troubleshoot with an IDE if the generated code is in the source directory.

File: pom.xml (hunk 1 above)
Comment (JEBailey) [reply pom.xml]: @rombert Some additional details on this. This bundle was using maven-bundle-plugin up until the prior release when there a desire to install a pax exam test and it was discovered there was a problem with the pax exam.  see https://issues.apache.org/jira/browse/SLING-7351. This particular bundle had a more complex solution then the one detailed in the link, with a bnd.bnd file specified and a change to bnd-maven-plugin. The maven-bundle-plugin provides a far better experience from a build perspective with a more detailed generation of a baseline report and with less configuration. So I took the suggested comment in that JIRA and updated this code accordingly

Comment (kwin) [reply pom.xml]: @JEBailey Actually SLING-7351 suggests to move to bnd-maven-plugin and not to stick to maven-bundle-plugin. What exactly was not working with bnd-maven-plugin. Basically we try to slowly move away from the old maven-bundle-plugin and only use the bnd-maven-plugin. Going back to the maven-bundle-plugin  just feels plain wrong to me.

Comment (JEBailey) [reply pom.xml]: @kwin  I misunderstood, from my experience between the two it feels like the bnd-maven-plugin is the step back. However to answer your question. Of the multiple concerns that I encountered, the specific problem is that the baselining is ignoring the package-info.java 

when I use bnd-maven-plugin I get the following
```
[INFO] --- bnd-baseline-maven-plugin:3.5.0:baseline (default) @ org.apache.sling.commons.html ---
[INFO] Automatically determining the baseline version for org.apache.sling:org.apache.sling.commons.html:jar:1.0.3-SNAPSHOT using repositories [apache.snapshots.https (https://repository.apache.org/content/repositories/snapshots, default, releases+snapshots), central (https://artifact.unx.sas.com:443/artifactory/es-maven-libs-release, default, releases), snapshots (https://artifact.unx.sas.com:443/artifactory/es-maven-libs-snapshot, default, releases+snapshots), apache.snapshots (https://repository.apache.org/snapshots, default, snapshots)]
[INFO] The baseline version was found to be 1.0.2
[ERROR] The bundle version change (1.0.2 to 1.0.3) is too low, the new version must be at least 1.1.0
```
It's obtaining the versioning information from the bundle instead of the package-info.

Comment (kwin) [reply pom.xml]: bnd-maven-plugin uses bnd under the hood as well. Please use the newest version (4.1.0) and compare those result against the ones from the latest maven-bundle-plugin. They should look very similar!

Comment (JEBailey) [reply pom.xml]: @kwin the baselining still looks different. I've gone ahead and created a new accessible branch based off of this one where I have it configured for bnd. It's still giving me the same message.

Comment (JEBailey) [reply pom.xml]: @kwin the branch I created for testing is https://github.com/apache/sling-org-apache-sling-commons-html/tree/BND-TEST. Since I'm unable to get the correct versioning I'll be staying with the sling-maven-plugin. Unless you have another suggestion for me to try.

Comment (kwin) [reply pom.xml]: @JEBailey I don't know either what is wrong with bnd in that case, but thanks a lot for the dedicated test branch. I created the ticket https://github.com/bndtools/bnd/issues/2859 to follow-up on that.

Comment (kwin) [reply pom.xml]: Actually bnd-baseline-maven-plugin is more correct than maven-bundle-plugin. Please compare with https://www.osgi.org/wp-content/uploads/SemanticVersioning.pdf:

> Requiring another bundle is similar to a short form of importing all the exported packages of that required
bundle. The version of a bundle must therefore semantically aggregate the semantics of all its constituent
packages. If any of these packages is incompatible with its providers then the bundle version must increment
the minor version. If any of these packages is incompatible with consumers, the bundle version must
increment the major version. It is clear, that on average, the version of a bundle will be much more volatile
than the versions of its constituent packages, increasing the dependency problems.
